### PR TITLE
feat(reuse): REUSE best practices for licensing/copyright part-3

### PR DESCRIPTION
**Pull request by rohitpandey49**
Signed-off-by: Rohit Pandey <rohit.pandey4900@gmail.com>

## Description

REUSE best practices for licensing/copyright

### Changes

1. Implemented correct licensing/copyright format
2. Used dep5 for `testdata` files

### Current REUSE status

When running reuse lint over this branch, the result is as follows:

```
* Files with copyright information: 4152 / 4360
* Files with license information: 3723 / 4360
```

Before that we were at:

```
* Files with copyright information: 4038 / 4360
* Files with license information: 3157 / 4360
```

<a href="https://gitpod.io/#https://github.com/fossology/fossology/pull/2248"></a>

### CODE REVIEW

File: src/srcdocs/makedox.license [hunk 1]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: © Fossology contributors
```
Comment (GMishx) [inline src/srcdocs/makedox.license:1]: File makedox looks like a bash script. So it will be better to put the licensing information there itself using `#`

File: src/testing/dataFiles/pkginstall/debian/6/apache2-php.ini [hunk 2]
```diff
@@ -1,5 +1,7 @@
 [PHP]
+; SPDX-FileCopyrightText: © Fossology contributors
 
+; SPDX-License-Identifier: GPL-2.0-only
```
Comment (GMishx) [inline src/testing/dataFiles/pkginstall/debian/6/apache2-php.ini:4]: Better license will be FSFAP for this file

File: src/testing/dataFiles/pkginstall/debian/6/cli-php.ini [hunk 3]
```diff
@@ -1,5 +1,7 @@
 [PHP]
+; SPDX-FileCopyrightText: © Fossology contributors
 
+; SPDX-License-Identifier: GPL-2.0-only
```
Comment (GMishx) [inline src/testing/dataFiles/pkginstall/debian/6/cli-php.ini:4]: Better license will be FSFAP for this file

File: src/testing/dataFiles/pkginstall/redhat/6.x/php.ini [hunk 4]
```diff
@@ -1,5 +1,7 @@
 [PHP]
+; SPDX-FileCopyrightText: © Fossology contributors
 
+; SPDX-License-Identifier: GPL-2.0-only
```
Comment (GMishx) [inline src/testing/dataFiles/pkginstall/redhat/6.x/php.ini:4]: Better license will be FSFAP for this file

File: src/testing/docker/test-cluster.sh [hunk 5]
```diff
@@ -1,3 +1,7 @@
+# SPDX-FileCopyrightText: © Fossology contributors
+
+# SPDX-License-Identifier: GPL-2.0-only
+
 #!/usr/bin/env bash
```
Comment (GMishx) [inline src/testing/docker/test-cluster.sh:5]: Please move this line to top of file.

File: src/testing/docker/test-standalone.sh [hunk 6]
```diff
@@ -1,3 +1,7 @@
+# SPDX-FileCopyrightText: © Fossology contributors
+
+# SPDX-License-Identifier: GPL-2.0-only
+
 #!/usr/bin/env bash
```
Comment (GMishx) [inline src/testing/docker/test-standalone.sh:5]: Please move this line to top of the file.

File: src/scheduler/ui_tests/scheduler_status.php [hunk 7]
```diff
@@ -1,8 +1,13 @@
...
+/*
+ SPDX-FileCopyrightText: © Fossology contributors
+
+ SPDX-License-Identifier: GPL-2.0-only
+*/
 /**
  * @file
  * @brief Test scheduler connection and status
- */
+**/
```
Comment (GMishx) [inline src/scheduler/ui_tests/scheduler_status.php:10]: Please revert unnecessary change

File: src/scheduler/agent/defconf/default/fossology [hunk 8]
```diff
@@ -1,9 +1,10 @@
+# SPDX-FileCopyrightText: © Fossology contributors
+
+# SPDX-License-Identifier: GPL-2.0-only
 # Defaults for fossology initscript
 # sourced by /etc/init.d/fossology
 
-#
 # This is a POSIX shell fragment
-#
```
Comment (GMishx) [inline src/scheduler/agent/defconf/default/fossology:6]: Please revert unnecessary change

**Review by GMishx — CHANGES_REQUESTED** [review]

File: src/srcdocs/makedox.license (hunk 1 above)
Comment (rohitpandey49) [reply src/srcdocs/makedox.license]: Done.

File: src/testing/dataFiles/pkginstall/debian/6/apache2-php.ini (hunk 2 above)
Comment (rohitpandey49) [reply src/testing/dataFiles/pkginstall/debian/6/apache2-php.ini]: Uodated.

File: src/testing/dataFiles/pkginstall/debian/6/cli-php.ini (hunk 3 above)
Comment (rohitpandey49) [reply src/testing/dataFiles/pkginstall/debian/6/cli-php.ini]: Updated.

File: src/testing/dataFiles/pkginstall/redhat/6.x/php.ini (hunk 4 above)
Comment (rohitpandey49) [reply src/testing/dataFiles/pkginstall/redhat/6.x/php.ini]: Updated.

File: src/testing/docker/test-cluster.sh (hunk 5 above)
Comment (rohitpandey49) [reply src/testing/docker/test-cluster.sh]: Updated.

File: src/scheduler/ui_tests/scheduler_status.php (hunk 7 above)
Comment (rohitpandey49) [reply src/scheduler/ui_tests/scheduler_status.php]: Corrected.

File: src/scheduler/agent/defconf/default/fossology (hunk 8 above)
Comment (rohitpandey49) [reply src/scheduler/agent/defconf/default/fossology]: Corrected.

File: src/testing/jenkins/Fossology_Build.xml [hunk 9]
```diff
@@ -1,3 +1,7 @@
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only 
+-->
```
Comment (GMishx) [inline src/testing/jenkins/Fossology_Build.xml:4]: Just noticed a behavior with XML files that if they do not start with `<?xml version='1.0' encoding='UTF-8'?>`, their mime-type changes.

To confirm, I changed `Fossology_Build.xml` as in this PR and did not touch the `Fossology_Functional_Tests.xml`. The mime-type for file where first line was a comment changed to `text/plain` rather than `text/xml`. This will cause some serious issues down the line.

Sorry for not catching it before, but can you please make sure all xml files start with the `<?xml?>` tag and comment can come on line 2?

```
$ file -i src/testing/jenkins/Fossology_Build.xml
src/testing/jenkins/Fossology_Build.xml: text/plain; charset=utf-8
$ file -i src/testing/jenkins/Fossology_Functional_Tests.xml 
src/testing/jenkins/Fossology_Functional_Tests.xml: text/xml; charset=us-ascii
```

File: src/wget_agent/agent/libwget_agent.a.license [hunk 10]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: © Fossology contributors
+
+SPDX-License-Identifier: GPL-2.0-only
```
Comment (GMishx) [inline src/wget_agent/agent/libwget_agent.a.license:3]: The file `src/wget_agent/agent/libwget_agent.a` is generated by the `src/wget_agent/agent/Makefile` and should ideally be added to the `.gitignore`

Comment (GMishx) [reply src/wget_agent/agent/libwget_agent.a.license:3]: See: https://github.com/fossology/fossology/blob/20ca3a482ce18e85859e92842e94fe06bb226237/.gitignore#L7

**Review by GMishx — CHANGES_REQUESTED** [review]

File: src/testing/jenkins/Fossology_Build.xml (hunk 9 above)
Comment (rohitpandey49) [reply src/testing/jenkins/Fossology_Build.xml:5]: Corrected.

File: src/ununpack/agent_tests/Functional/tests.xml [hunk 11]
```diff
@@ -1,3 +1,7 @@
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only
+-->
```
[the remark this answers is not in the document]
Comment (rohitpandey49) [reply src/ununpack/agent_tests/Functional/tests.xml:5]: Corrected.

File: src/wget_agent/agent/libwget_agent.a.license (hunk 10 above)
Comment (rohitpandey49) [reply src/wget_agent/agent/libwget_agent.a.license]: Corrected.

File: src/cli/tests/tests.xml [hunk 12]
```diff
@@ -1,3 +1,4 @@
+<?xml?>
 <!-- 
   SPDX-FileCopyrightText: © Fossology contributors
```
Comment (GMishx) [inline src/cli/tests/tests.xml:3]: Please remove the `<?xml?>` tag and move the comment to line after `<phpunit>`

File: src/lib/php/tests/tests.xml [hunk 13]
```diff
@@ -1,3 +1,4 @@
+<?xml?>
 <!-- SPDX-FileCopyrightText: © Fossology contributors
 
      SPDX-License-Identifier: GPL-2.0-only
```
Comment (GMishx) [inline src/lib/php/tests/tests.xml:4]: Please remove the `<?xml?>` tag and move the comment to line after `<phpunit>`

File: src/nomos/agent_tests/Functional/tests.xml [hunk 14]
```diff
@@ -1,3 +1,4 @@
+<?xml?>
 <!-- SPDX-FileCopyrightText: © Fossology contributors
 
      SPDX-License-Identifier: FSFAP
```
Comment (GMishx) [inline src/nomos/agent_tests/Functional/tests.xml:4]: Please remove the `<?xml?>` tag and move the comment to line after `<phpunit>`

File: src/testing/jenkins/Fossology_Build.xml [hunk 15]
```diff
@@ -1,3 +1,8 @@
+<?xml?>
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only 
+-->
 <?xml version='1.0' encoding='UTF-8'?>
 <project>
```
Comment (GMishx) [inline src/testing/jenkins/Fossology_Build.xml:7]:
```suggestion
<?xml version='1.0' encoding='UTF-8'?>
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only 
-->
<project>
```

File: src/testing/jenkins/Fossology_Functional_Tests.xml [hunk 16]
```diff
@@ -1,3 +1,8 @@
+<?xml?>
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only 
+-->
 <?xml version='1.0' encoding='UTF-8'?>
 <project>
```
Comment (GMishx) [inline src/testing/jenkins/Fossology_Functional_Tests.xml:7]:
```suggestion
<?xml version='1.0' encoding='UTF-8'?>
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only 
-->
<project>
```

File: src/testing/jenkins/Fossology_Installation.xml [hunk 17]
```diff
@@ -1,3 +1,8 @@
+<?xml?>
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only 
+-->
 <?xml version='1.0' encoding='UTF-8'?>
 <project>
```
Comment (GMishx) [inline src/testing/jenkins/Fossology_Installation.xml:7]:
```suggestion
<?xml version='1.0' encoding='UTF-8'?>
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only 
-->
<project>
```

File: src/testing/jenkins/Fossology_Migration.xml [hunk 18]
```diff
@@ -1,3 +1,8 @@
+<?xml?>
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only 
+-->
 <?xml version='1.0' encoding='UTF-8'?>
 <matrix-project>
```
Comment (GMishx) [inline src/testing/jenkins/Fossology_Migration.xml:7]:
```suggestion
<?xml version='1.0' encoding='UTF-8'?>
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only 
-->
<matrix-project>
```

File: src/testing/jenkins/Fossology_Package_Build.xml [hunk 19]
```diff
@@ -1,3 +1,8 @@
+<?xml?>
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only 
+-->
 <?xml version='1.0' encoding='UTF-8'?>
 <project>
```
Comment (GMishx) [inline src/testing/jenkins/Fossology_Package_Build.xml:7]:
```suggestion
<?xml version='1.0' encoding='UTF-8'?>
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only 
-->
<project>
```

File: src/testing/jenkins/Fossology_Package_Install.xml [hunk 20]
```diff
@@ -1,3 +1,8 @@
+<?xml?>
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only 
+-->
 <?xml version='1.0' encoding='UTF-8'?>
 <matrix-project>
```
Comment (GMishx) [inline src/testing/jenkins/Fossology_Package_Install.xml:7]:
```suggestion
<?xml version='1.0' encoding='UTF-8'?>
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only 
-->
<matrix-project>
```

File: src/testing/jenkins/Fossology_Package_Install_Prep.xml [hunk 21]
```diff
@@ -1,3 +1,8 @@
+<?xml?>
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only 
+-->
 <?xml version='1.0' encoding='UTF-8'?>
 <project>
```
Comment (GMishx) [inline src/testing/jenkins/Fossology_Package_Install_Prep.xml:7]:
```suggestion
<?xml version='1.0' encoding='UTF-8'?>
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only 
-->
<project>
```

File: src/testing/jenkins/Fossology_Performance.xml [hunk 22]
```diff
@@ -1,3 +1,8 @@
+<?xml?>
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only 
+-->
 <?xml version='1.0' encoding='UTF-8'?>
 <project>
```
Comment (GMishx) [inline src/testing/jenkins/Fossology_Performance.xml:7]:
```suggestion
<?xml version='1.0' encoding='UTF-8'?>
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only 
-->
<project>
```

File: src/testing/jenkins/Fossology_Performance_Prep.xml [hunk 23]
```diff
@@ -1,3 +1,8 @@
+<?xml?>
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only 
+-->
 <?xml version='1.0' encoding='UTF-8'?>
 <project>
```
Comment (GMishx) [inline src/testing/jenkins/Fossology_Performance_Prep.xml:7]:
```suggestion
<?xml version='1.0' encoding='UTF-8'?>
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only 
-->
<project>
```

File: src/testing/jenkins/Fossology_Source_Install.xml [hunk 24]
```diff
@@ -1,3 +1,8 @@
+<?xml?>
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only 
+-->
 <?xml version='1.0' encoding='UTF-8'?>
 <matrix-project>
```
Comment (GMishx) [inline src/testing/jenkins/Fossology_Source_Install.xml:7]:
```suggestion
<?xml version='1.0' encoding='UTF-8'?>
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only 
-->
<matrix-project>
```

File: src/testing/jenkins/Fossology_Unit_Test_Prep.xml [hunk 25]
```diff
@@ -1,3 +1,8 @@
+<?xml?>
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only 
+-->
 <?xml version='1.0' encoding='UTF-8'?>
 <project>
```
Comment (GMishx) [inline src/testing/jenkins/Fossology_Unit_Test_Prep.xml:7]:
```suggestion
<?xml version='1.0' encoding='UTF-8'?>
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only 
-->
<project>
```

File: src/testing/jenkins/Fossology_Unit_Tests.xml [hunk 26]
```diff
@@ -1,3 +1,8 @@
+<?xml?>
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only 
+-->
 <?xml version='1.0' encoding='UTF-8'?>
 <project>
```
Comment (GMishx) [inline src/testing/jenkins/Fossology_Unit_Tests.xml:7]:
```suggestion
<?xml version='1.0' encoding='UTF-8'?>
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only 
-->
<project>
```

File: src/testing/jenkins/Fossology_Unit_Tests_Cleanup.xml [hunk 27]
```diff
@@ -1,3 +1,8 @@
+<?xml?>
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only 
+-->
 <?xml version='1.0' encoding='UTF-8'?>
 <project>
```
Comment (GMishx) [inline src/testing/jenkins/Fossology_Unit_Tests_Cleanup.xml:7]:
```suggestion
<?xml version='1.0' encoding='UTF-8'?>
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only 
-->
<project>
```

File: src/testing/jenkins/Generate_Code_Documentation.xml [hunk 28]
```diff
@@ -1,3 +1,8 @@
+<?xml?>
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only 
+-->
 <?xml version='1.0' encoding='UTF-8'?>
 <project>
```
Comment (GMishx) [inline src/testing/jenkins/Generate_Code_Documentation.xml:7]:
```suggestion
<?xml version='1.0' encoding='UTF-8'?>
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only 
-->
<project>
```

File: src/ununpack/agent_tests/Functional/tests.xml [hunk 29]
```diff
@@ -1,3 +1,8 @@
+<?xml?>
+<!-- SPDX-FileCopyrightText: © Fossology contributors
+
+     SPDX-License-Identifier: GPL-2.0-only
+-->
 <phpunit>
   <testsuites>
```
Comment (GMishx) [inline src/ununpack/agent_tests/Functional/tests.xml:7]:
```suggestion
<phpunit>
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only
-->
  <testsuites>
```

File: src/www/ui/css/select2.min.css [hunk 30]
```diff
@@ -1 +1,7 @@
+/*
+ SPDX-FileCopyrightText: © Fossology contributors
+
+ SPDX-License-Identifier: GPL-2.0-only
```
Comment (GMishx) [inline src/www/ui/css/select2.min.css:4]: Please change license to MIT and add following copyrights:
`Copyright (c) 2012-2017 Kevin Brown, Igor Vaynberg, and Select2 contributors`

The file is licensed here: https://github.com/select2/select2/blob/master/LICENSE.md

**Review by GMishx — CHANGES_REQUESTED** [review]

File: src/nomos/agent/GenCodeCopyright [hunk 31]
```diff
@@ -1,17 +1,5 @@
...
- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- GNU General Public License for more details.
-
- You should have received a copy of the GNU General Public License along
- with this program; if not, write to the Free Software Foundation, Inc.,
- 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
-
- ***************************************************************
+ SPDX-License-Identifier: GPL-2.0-only
+*/
```
Comment (GMishx) [inline src/nomos/agent/GenCodeCopyright:5]: Please remove the `/*` and `*/` from the file content.

The content of the file gets copied to other files in format:
```c
/*
 * <Content of this file>
 */
```

The `*/` causes them to be:
```c
/*
 * /*
 SPDX-FileCopyrightText: © 2006-2009 Hewlett-Packard Development Company, L.P.

SPDX-License-Identifier: GPL-2.0-only
*/
 */
```

And it is causing code compilation failures.

**Review by GMishx — CHANGES_REQUESTED** [review]

**Comment by rohitpandey49** [comment]
Corrected. Please check @GMishx

File: src/nomos/agent_tests/Functional/tests.xml (hunk 14 above)
Comment (GMishx) [reply src/nomos/agent_tests/Functional/tests.xml:4]: Reminder @rohitpandey49

File: src/testing/jenkins/Fossology_Package_Install.xml (hunk 20 above)
Comment (GMishx) [reply src/testing/jenkins/Fossology_Package_Install.xml:7]: Reminder @rohitpandey49

File: src/testing/jenkins/Fossology_Source_Install.xml (hunk 24 above)
Comment (GMishx) [reply src/testing/jenkins/Fossology_Source_Install.xml:7]: Reminder @rohitpandey49

File: src/testing/jenkins/Fossology_Unit_Tests.xml (hunk 26 above)
Comment (GMishx) [reply src/testing/jenkins/Fossology_Unit_Tests.xml:7]: Reminder @rohitpandey49

**Comment by GMishx** [comment]
Hello @rohitpandey49 , there are 5 suggestions remaining to be implemented. I have commented on them as a reminder.

Please fix them so we can merge the PR.

**Comment by rohitpandey49** [comment]
> Hello @rohitpandey49 , there are 5 suggestions remaining to be implemented. I have commented on them as a reminder.
> 
> Please fix them so we can merge the PR.

Fixed @GMishx sir

**Review by GMishx — APPROVED** [review]
Changes looks good.
